openfeature/telemetry/telemetry.go

```diff
@@ -5,96 +5,64 @@ import (
 	"strings"
 
 	"github.com/open-feature/go-sdk/openfeature"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.34.0"
 )
 
-// EvaluationEvent represents an event that is emitted when a flag is evaluated.
-// It is intended to be used to record flag evaluation events as OpenTelemetry log records.
-// See the OpenFeature specification [Appendix D: Observability] and
-// the OpenTelemetry [Semantic conventions for feature flags in logs] for more information.
-//
-// [Appendix D: Observability]: https://openfeature.dev/specification/appendix-d
-// [Semantic conventions for feature flags in logs]: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
-type EvaluationEvent struct {
-	// Name is the name of the event.
-	// It is always "feature_flag.evaluation".
-	Name string
-	// Attributes represents the event's attributes.
-	Attributes map[string]any
-}
-
-// The OpenTelemetry compliant event attributes for flag evaluation.
-const (
-	FlagKey          string = "feature_flag.key"
-	ErrorTypeKey     string = "error.type"
-	ResultValueKey   string = "feature_flag.result.value"
-	ResultVariantKey string = "feature_flag.result.variant"
-	ErrorMessageKey  string = "error.message"
-	ContextIDKey     string = "feature_flag.context.id"
-	ProviderNameKey  string = "feature_flag.provider.name"
-	ResultReasonKey  string = "feature_flag.result.reason"
-	FlagSetIDKey     string = "feature_flag.set.id"
-	VersionKey       string = "feature_flag.version"
-)
-
-// FlagEvaluationKey is the name of the feature flag evaluation event.
-const FlagEvaluationKey string = "feature_flag.evaluation"
+// EventName is the name of the feature flag evaluation event.
+const EventName string = "feature_flag.evaluation"
 
 const (
 	flagMetaContextIDKey string = "contextId"
 	flagMetaFlagSetIDKey string = "flagSetId"
 	flagMetaVersionKey   string = "version"
 )
 
-// CreateEvaluationEvent creates an [EvaluationEvent].
+// EventAttributes returns a slice of OpenTelemetry attributes that can be used to create an event for a feature flag evaluation.
 // It is intended to be used in the `Finally` stage of a [openfeature.Hook].
-func CreateEvaluationEvent(hookContext openfeature.HookContext, details openfeature.InterfaceEvaluationDetails) EvaluationEvent {
-	attributes := map[string]any{
-		FlagKey:         hookContext.FlagKey(),
-		ProviderNameKey: hookContext.ProviderMetadata().Name,
+func EventAttributes(hookContext openfeature.HookContext, details openfeature.InterfaceEvaluationDetails) []attribute.KeyValue {
```

Comment:
Since this is a new function, should we keep the old one with a deprecation warning? I don't want technical debt to pile up, but this seems like an easy way to avoid abruptly adding a breaking change.

Reply:
We could branch our `v1` now and this could be a breaking change, but I think we should consider [this](https://github.com/open-feature/go-sdk/pull/418/files#r2245958523).

If we want to implement this in contribs instead, to avoid the OTel dep (which would be my preference) I'd recommend we just cut the `v1` maintenance branch now, and then completely remove this function here and implement it in contribs.

WDYT @beeme1mr @sahidvelji ?